Tools/Create-Azure-Sentinel-Solution/common/createCCPConnector.ps1

```diff
@@ -0,0 +1,880 @@
+    {
+        $parentIdResourceName = "'Microsoft.SecurityInsights/dataConnectorDefinitions'"
+    }
+    else {
+        $parentIdResourceName = "'Microsoft.SecurityInsights/dataConnectors'"
+    }
+
+    $parentId = "[extensionResourceId(resourceId('Microsoft.OperationalInsights/workspaces', parameters('workspace')), $parentIdResourceName, variables('_dataConnectorContentId$($templateKindByCounter[$TemplateCounter])$($global:connectorCounter)'))]"
+    $metaDataResourceName = "concat('DataConnector-', variables('_dataConnectorContentId$($templateKindByCounter[$TemplateCounter])$($global:connectorCounter)'))"
+    $metaDataContentId = "[variables('_dataConnectorContentId$($templateKindByCounter[$TemplateCounter])$($global:connectorCounter)')]"
+    $metaDatsContentVersion  = "[variables('dataConnectorVersion$($templateKindByCounter[$TemplateCounter])')]"
+    $metaDataResource =  Get-MetaDataBaseResource $metaDataResourceName $parentId $metaDataContentId $templateContentTypeByCounter[$TemplateCounter] $metaDatsContentVersion $dataFileMetadata $solutionFileMetadata
+    
+    if($templateContentTypeByCounter[$TemplateCounter] -eq "DataConnector")
+    {
+        $dependencies = [PSCustomObject]@{
+                "criteria" =  @(
+                    [PSCustomObject]@{
+                        "version" = "[variables('dataConnectorVersion$($templateKindByCounter[2])')]";
+                        "contentId" = "[variables('_dataConnectorContentId$($templateKindByCounter[2])$($global:connectorCounter)')]";
+                        "kind" = "ResourcesDataConnector"
+                    }
+                )
+            }
+
+        $metaDataResource.properties | Add-Member -NotePropertyName "dependencies" -NotePropertyValue $dependencies
+    }
+
+    return  $metaDataResource;
+}
+
+function Get-ContentTemplateResource($contentResourceDetails, $TemplateCounter, $ccpItem){
+    $contentVersion = "variables('dataConnectorVersion$($templateKindByCounter[$TemplateCounter])')";
+    $contentTemplateName = "variables('dataConnectorTemplateName$($templateKindByCounter[$TemplateCounter])$($global:connectorCounter)')";
+    $contentId = "variables('_dataConnectorContentId$($templateKindByCounter[$TemplateCounter])$($global:connectorCounter)')";
+    $resoureKind = $templateContentTypeByCounter[$TemplateCounter];
+    if($resoureKind -eq "DataConnector")
+    {
+        $resoureKindTag = "dc";
+    }
+    else {
+        $resoureKindTag = "rdc";
+    }
+
+    $title = $ccpItem.title;
+    $displayName = $title;
+    $randomNumber = Get-Random
+    return [PSCustomObject]@{
+        type       = "Microsoft.OperationalInsights/workspaces/providers/contentTemplates";
+        apiVersion = $contentResourceDetails.metadataApiVersion; # "2023-04-01-preview";
+        name        = "[concat(parameters('workspace'),'/Microsoft.SecurityInsights/', $contentTemplateName, $randomNumber)]";
+        location   = "[parameters('workspace-location')]";
```

Comment:
location is needed?

Reply:
yes, location is needed. If we remove then if fails in deployment.
![image](https://github.com/Azure/Azure-Sentinel/assets/107389644/f73fe659-4fd0-406f-a2df-da2471da115c)